### PR TITLE
Cloud API: add support for container embedding

### PR DESCRIPTION
**Pull request by gicmo**
Add support for embedding container images via the cloud API. For this the container resolve job was plumbed into the cloud api's handler and the API specification updated with a new `containers`section that mimics the blueprint section with the same name.

### CODE REVIEW

File: internal/cloudapi/v2/server.go [hunk 1]
```diff
@@ -164,7 +191,33 @@ func (s *Server) enqueueKojiCompose(taskID uint64, server, name, version, releas
 			return id, HTTPErrorWithInternal(ErrorEnqueueingJob, err)
 		}
 
-		manifestJobID, err := s.workers.EnqueueManifestJobByID(&worker.ManifestJobByID{}, depsolveJobID, channel)
+		var containerResolveJob uuid.UUID
+		dependencies := []uuid.UUID{depsolveJobID}
+		if len(bp.Containers) > 0 {
+			job := worker.ContainerResolveJob{
```
[the remark this answers is not in the document]
Comment (gicmo) [reply internal/cloudapi/v2/server.go:197]: I was tempted by this already but then none of the surrounding code is shared yet either, so I refrained from too much refactoring. But maybe we should do that for all of it in a follow up.

[the remark this answers is not in the document]
Comment (ygalblum) [reply internal/cloudapi/v2/server.go:197]: Yeah, you're right. The code has some huge methods

File: test/cases/api/container.registry.sh [hunk 2]
```diff
@@ -47,8 +48,14 @@ function createReqFile() {
...
       }
     ],
+    "containers":[
+      {
+        "source": "$CONTAINER_SOURCE"
+      }
+    ],
     "packages": [
       "postgresql",
+      "podman",
```
Comment (thozza) [inline test/cases/api/container.registry.sh:58]: I wonder whether we should not consider adding some container runtime to the package set automatically if the `containers` customization is specified 🤔

**Review by thozza — COMMENTED** [review]
One issue: when you rebased on top of `main`, you fixed the code to adapt for the changes in `clienterrors.WorkerClientError` only in the second commit. As a result, the code would not compile with only the first commit.

The rest looks good 👍

**Review by thozza — APPROVED** [review]
Fabulous 🤩

File: test/cases/api/container.registry.sh (hunk 2 above)
Comment (gicmo) [reply test/cases/api/container.registry.sh:58]: Yeah, that is a good idea; I guess we should introduce a customisation for it like we do with the kernel, where one can select the desired container runtime `["podman", "cri-o", ... "docker"].

**Pull request merged**